File: devtools/platform-descriptor-json/src/main/resources/bundled-codestarts/example/commandmode-example/java/src/main/java/org/acme/commandmode/GreetingMain.tpl.qute.java
package org.acme.commandmode;

import javax.enterprise.context.control.ActivateRequestContext;
import javax.inject.Inject;

import io.quarkus.runtime.Quarkus;
import io.quarkus.runtime.QuarkusApplication;
import io.quarkus.runtime.annotations.QuarkusMain;

@QuarkusMain
public class GreetingMain implements QuarkusApplication {

    @Inject
    GreetingService service;

    @Override
    public int run(String... args) throws Exception {

        if(args.length>0) {
            System.out.println(service.greeting(String.join(" ", args)));
        } else {
            System.out.println(service.greeting("{greeting.default-name}"));
        }

        return 0;
    }
}

File: devtools/platform-descriptor-json/src/main/resources/bundled-codestarts/example/commandmode-example/kotlin/src/main/kotlin/org/acme/commandmode/GreetingMain.tpl.qute.kt
package org.acme.commandmode

import javax.inject.Inject

import io.quarkus.runtime.QuarkusApplication
import io.quarkus.runtime.annotations.QuarkusMain

@QuarkusMain
class GreetingMain: QuarkusApplication {

    @Inject
    lateinit var service: GreetingService
    override fun run(vararg args: String?): Int {
        if (args.isNotEmpty()) {
            System.out.println(service.greeting(args.joinToString(",")))
        } else {
            System.out.println(service.greeting("{greeting.default-name}"))
        }
        return 0
    }

}
